### PR TITLE
fix: fixed bug in error handling in experiment.go

### PR DESCRIPTION
**Pull request by salonig23**
## Description

Missed `err == nil` condition in a `handleContinueExperiment` function in a previous ticket. Fixed it by adding the if condition.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/internal/experiment.go [hunk 1]
```diff
@@ -764,7 +766,6 @@ func (e *internalExperiment) processOperations(
 				e.syslog.Errorf("trial %s already exists", op.RequestID)
 				continue
 			}
-
```
Comment (NicholasBlaskey) [inline master/internal/experiment.go:767]: unneeded whitespace change?

File: master/internal/experiment.go [hunk 2]
```diff
@@ -727,11 +727,13 @@ func (e *internalExperiment) handleContinueExperiment(reqID model.RequestID) (*i
 			e.syslog.Error(err)
 			return nil, true
 		}
-		if trial.State != model.CompletedState {
```
Comment (NicholasBlaskey) [inline master/internal/experiment.go:730]: this code feels a little complicated to me now

maybe something like a switch statement could make this more clear we have three cases going on

```
switch {
case errors.Is(err, sql.ErrNoRows):
    // Trial doesn't exist, don't do anything
case err != nil:
   e.updateState ... 
case err == nil:
   if state...
}
```

Comment (salonig23) [reply master/internal/experiment.go:730]: Done!

**Review by NicholasBlaskey — APPROVED** [review]
lgtm nice work

**Pull request merged**